src/uu/ln/src/ln.rs

```diff
@@ -277,8 +284,36 @@ fn link_files_in_dir(files: &[PathBuf], target_dir: &PathBuf, settings: &Setting
     }
 }
 
+fn relative_path(src: &PathBuf, dst: &PathBuf) -> PathBuf {
+    let abssrc = canonicalize(src, CanonicalizeMode::Normal).unwrap();
+    let absdst = canonicalize(dst, CanonicalizeMode::Normal).unwrap();
```

Comment:
This function should return `std::io::Result<PathBuf>`.  These `unwrap()`s should just return the error from `canonicalize()`.